src/library_syscall.js

```diff
@@ -679,8 +679,7 @@ var SyscallsLibrary = {
     return SYSCALLS.doStat(FS.lstat, path, buf);
   },
   __syscall_fstat64: (fd, buf) => {
-    var stream = SYSCALLS.getStreamFromFD(fd);
-    return SYSCALLS.doStat(FS.stat, stream.path, buf);
+    return SYSCALLS.doStat(FS.fstat, fd, buf);
```

Comment:
Perhaps rename the second argument to `doStat` if we are going to allow it to take things that are not paths?